travel-agency-persistence/src/test/java/cz/fi/muni/pa165/travelagency/dao/ReservationDaoTest.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package cz.fi.muni.pa165.travelagency.dao;
+
+import cz.fi.muni.pa165.travelagency.EmbeddedDatabaseSpringConfig;
+import cz.fi.muni.pa165.travelagency.entity.Customer;
+import cz.fi.muni.pa165.travelagency.entity.Reservation;
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceContext;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.TestExecutionListeners;
+import org.springframework.test.context.support.DependencyInjectionTestExecutionListener;
+import org.springframework.test.context.support.DirtiesContextTestExecutionListener;
+import org.springframework.test.context.testng.AbstractTestNGSpringContextTests;
+import org.springframework.test.context.transaction.TransactionalTestExecutionListener;
+import org.springframework.transaction.annotation.Transactional;
+import org.testng.Assert;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+/**
+ *
+ * @author Diana Vilkolakova
+ */
+@ContextConfiguration(classes = EmbeddedDatabaseSpringConfig.class)
+@TestExecutionListeners(inheritListeners = false, listeners = { DependencyInjectionTestExecutionListener.class, DirtiesContextTestExecutionListener.class, TransactionalTestExecutionListener.class })
+@Transactional
+public class ReservationDaoTest  extends AbstractTestNGSpringContextTests{
+    @PersistenceContext
+    private EntityManager em;
+    
+    @Autowired
+    private ReservationDao reservationDao;
+    
+    private Reservation reservation;
+    private Customer customer;
+    private final Long price = 5000l;
+    
+    @BeforeMethod
+    public void createReservations() {
+        reservation = new Reservation();
+        customer = new Customer();
+        
+        customer.setEmail("diana@email.com");
+        customer.setFirstName("Diana");
+        customer.setLastName("Vilko");
+        customer.setPassword("password");
+        customer.setUsername("dada");
+        
+        reservation.setCustomer(customer);
+        reservation.setPrice(price);
```

Comment:
Hod do tej rezervacie potom aj nejake tripy, ked implementujes metodu addTrip. V teste to potom over, ze tam su.